tst/EnergyPlus/unit/SolarShading.unit.cc

```diff
@@ -3476,3 +3477,405 @@ TEST_F(EnergyPlusFixture, SolarShadingTest_CTRANS)
+        "    0.2,                     !- Diffuse Solar Reflectance of Unglazed Part of Shading",
+        "    0.2;                     !- Diffuse Visible Reflectance of Unglazed Part of Shading",
+
+        "  Shading:Site:Detailed,",
+        "    Zn001:Wall-South:Shade002,  !- Name",
+        "    SunShading,              !- Transmittance Schedule Name",
+        "    4,                       !- Number of Vertices",
+        "    -3,-5,5,               !- X,Y,Z ==> Vertex 1 {m}",
+        "    -3,-6,5,               !- X,Y,Z ==> Vertex 2 {m}",
+        "    3,-6,5,                !- X,Y,Z ==> Vertex 3 {m}",
+        "    3,-5,5;                !- X,Y,Z ==> Vertex 4 {m}",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    SimulationManager::GetProjectData(*state);
+    bool FoundError = false;
+
+    HeatBalanceManager::GetProjectControlData(*state, FoundError); // read project control data
+    EXPECT_FALSE(FoundError);                                      // expect no errors
+
+    HeatBalanceManager::SetPreConstructionInputParameters(*state);
+    ScheduleManager::ProcessScheduleInput(*state); // read schedules
+
+    HeatBalanceManager::GetMaterialData(*state, FoundError);
+    EXPECT_FALSE(FoundError);
+
+    HeatBalanceManager::GetFrameAndDividerData(*state, FoundError);
+    EXPECT_FALSE(FoundError);
+
+    HeatBalanceManager::GetConstructData(*state, FoundError);
+    EXPECT_FALSE(FoundError);
+
+    HeatBalanceManager::GetZoneData(*state, FoundError);
+    EXPECT_FALSE(FoundError);
+
+    SurfaceGeometry::GetGeometryParameters(*state, FoundError);
+    EXPECT_FALSE(FoundError);
+
+    state->dataSurfaceGeometry->CosZoneRelNorth.allocate(1);
+    state->dataSurfaceGeometry->SinZoneRelNorth.allocate(1);
+
+    state->dataSurfaceGeometry->CosZoneRelNorth(1) = std::cos(-state->dataHeatBal->Zone(1).RelNorth * DataGlobalConstants::DegToRadians);
+    state->dataSurfaceGeometry->SinZoneRelNorth(1) = std::sin(-state->dataHeatBal->Zone(1).RelNorth * DataGlobalConstants::DegToRadians);
+    state->dataSurfaceGeometry->CosBldgRelNorth = 1.0;
+    state->dataSurfaceGeometry->SinBldgRelNorth = 0.0;
+
+    // This function will call SurfaceGeometry::GetDetShdSurfaceData() and SurfaceGeometry::GetAttShdSurfaceData(), which are the target functions to
+    // be tested
+    SurfaceGeometry::GetSurfaceData(*state, FoundError);
+    EXPECT_FALSE(FoundError);
+
+    EXPECT_EQ(state->dataErrTracking->TotalWarningErrors, 1);
+    // Expect to have two severre errors, one for the detached Shading:Site:Detailed, and one for the attached Shading:Zone:Detailed
+    EXPECT_EQ(state->dataErrTracking->TotalSevereErrors, 2);
+    // The attached one will be the last severe error, since the "attached" function is called later than the "detached" one.
+    EXPECT_EQ(state->dataErrTracking->LastSevereError,
+              "Shading:Zone:Detailed=\"ZN001:WALL-SOUTH:SHADE001\" has an active Transmittance Schedule Name=\"SUNSHADING\";");
+
```

Comment:
Expect to have two severe errors, one for the detached `Shading:Site:Detailed`, and one for the attached `Shading:Zone:Detailed`. 

The attached one will be the last severe error, since the "attached" function is called later than the "detached" one.

Reply:
With the new method implemented, there is only one severe message expected and it should come from `SolarShading::GetShadowingInput()`.